modules/core/src/test/java/org/apache/ignite/internal/ClusterNodeMetricsSelfTest.java

```diff
@@ -75,7 +96,84 @@
         cfg.setCacheConfiguration();
         cfg.setMetricsUpdateFrequency(0);
 
-        return cfg;
+        CacheConfiguration<Integer, Object> offHeapValuesCfg = defaultCacheConfiguration();
+        offHeapValuesCfg.setName(OFF_HEAP_VALUE_NAME);
+        offHeapValuesCfg.setStatisticsEnabled(true);
+        offHeapValuesCfg.setMemoryMode(CacheMemoryMode.OFFHEAP_VALUES);
+        offHeapValuesCfg.setOffHeapMaxMemory(MAX_VALS_AMOUNT * VAL_SIZE);
+
+        CacheConfiguration<Integer, Object> onHeapTieredCfg = defaultCacheConfiguration();
+        onHeapTieredCfg.setName(ON_HEAP_TIERED_NAME);
+        onHeapTieredCfg.setStatisticsEnabled(true);
+        onHeapTieredCfg.setMemoryMode(CacheMemoryMode.ONHEAP_TIERED);
+        onHeapTieredCfg.setOffHeapMaxMemory(MAX_VALS_AMOUNT * VAL_SIZE);
+
+        FifoEvictionPolicy plc = new FifoEvictionPolicy();
+        plc.setMaxMemorySize(MAX_VALS_AMOUNT * VAL_SIZE);
+        plc.setMaxSize(0);
+
+        onHeapTieredCfg.setEvictionPolicy(plc);
+
+        return cfg.setCacheConfiguration(offHeapValuesCfg, onHeapTieredCfg);
+    }
+
+    /**
+     *
+     */
+    public void testAllocatedMemory() throws Exception {
+        Ignite ignite = grid();
+
+        final IgniteCache onHeapCache = ignite.getOrCreateCache(ON_HEAP_TIERED_NAME);
+        final IgniteCache offHeapCache = ignite.getOrCreateCache(OFF_HEAP_VALUE_NAME);
+
+        long prevTieredOffHeapSize = onHeapCache.metrics().getOffHeapAllocatedSize();
+        long prevValuesOffHeapSize = offHeapCache.metrics().getOffHeapAllocatedSize();
+
+        assertEquals(0, prevTieredOffHeapSize);
+        assertEquals(0, prevValuesOffHeapSize);
+
+        long prevClusterNonHeapMemoryUsed = ignite.cluster().metrics().getNonHeapMemoryUsed();
+
+        fillCache(onHeapCache, getTestTimeout());
+
+        assertTrue(onHeapCache.metrics().getOffHeapAllocatedSize() > (MAX_VALS_AMOUNT - 5) * VAL_SIZE + prevTieredOffHeapSize);
+        assertEquals(0, offHeapCache.metrics().getOffHeapAllocatedSize());
+
+        // prevClusterNonHeapMemoryUsed = 69_277_368
+        // (MAX_VALS_AMOUNT - 5) * VAL_SIZE = 49_807_360
+        // tiered.metrics().getOffHeapAllocatedSize() = 51_012_531
+
+        assertTrue(prevClusterNonHeapMemoryUsed < ignite.cluster().metrics().getNonHeapMemoryUsed());
+
+        prevClusterNonHeapMemoryUsed = ignite.cluster().metrics().getNonHeapMemoryUsed();
+        prevTieredOffHeapSize = onHeapCache.metrics().getOffHeapAllocatedSize();
+
+        fillCache(offHeapCache, getTestTimeout());
+
+        assertTrue(offHeapCache.metrics().getOffHeapAllocatedSize() > (MAX_VALS_AMOUNT - 5) * VAL_SIZE);
+        assertEquals(prevTieredOffHeapSize, onHeapCache.metrics().getOffHeapAllocatedSize());
+        assertTrue((MAX_VALS_AMOUNT - 5) * VAL_SIZE + prevClusterNonHeapMemoryUsed < ignite.cluster().metrics().getNonHeapMemoryUsed());
+    }
+
+    /** Fill cache with values. */
```

Comment:
Documentation doesn't align with our guidelines. See how the documentation looks like in the tests.